flocker/node/agents/functional/test_ebs.py

```diff
@@ -156,3 +160,273 @@ def test_next_device_in_use(self):
         result = self.api._next_device(self.api.compute_instance_id(), [],
                                        {u"/dev/sdf"})
         self.assertEqual(result, u"/dev/sdg")
+
+
+class VolumeStateTransitionTests(TestCase):
+    """
+    Tests for volume state operations and resulting volume state changes.
+    """
+
+    def _create_template_ebs_volume(self, operation):
+        """
+        Helper function to create template EBS volume to work on.
+
+        :param NamedConstant operation: Intended use of created template.
+            A value from ``VolumeOperations``.
+
+        :returns: Suitable volume in the right start state for input operation.
+        :rtype: boto.ec2.volume.Volume
+        """
+        volume = EbsVolume()
+
+        # Irrelevant volume attributes.
+        volume.id = u'vol-9c48a689'
+        volume.create_time = u'2015-07-14T22:46:00.447Z'
+        volume.size = 1
+        volume.snapshot_id = ''
+        volume.zone = u'us-west-2b'
+        volume.type = u'standard'
+
+        volume_state_table = VolumeStateTable(table=populate())
+        state_flow = volume_state_table.table[operation]
+        start_state = state_flow.start_state.value
+
+        # Interesting volume attribute.
+        volume.status = start_state
+
+        return volume
+
+    def _pick_invalid_state(self, operation):
+        """
+        Helper function to pick an invalid volume state for input operation.
+
+        :param NamedConstant operation: Volume operation to pick a state for.
+            A value from ``VolumeOperations``.
+
+        :returns: A state from ``VolumeStates`` that will not be part of
+            a volume's states resulting from input operation.
+        :rtype: ValueConstant
+        """
+        volume_state_table = VolumeStateTable(table=populate())
+        state_flow = volume_state_table.table[operation]
+        valid_states = set([state_flow.start_state,
+                            state_flow.transient_state,
+                            state_flow.end_state])
+        invalid_states = set(VolumeStates._enumerants.values()) - valid_states
+        invalid_state = invalid_states.pop()
+        if invalid_state is None:
```

Comment:
It doesn't seem like `None` should ever happen:

1. `pop()` will throw KeyError I suspect if set is empty.
2. Logically this oughtn't happen since number of states is > 3.

So you can just skip the `None` code path.